src/secret/ledger_nano.rs

```diff
@@ -95,160 +114,217 @@ impl SecretManage for LedgerSecretManager {
+                        s.native_tokens().as_ref(),
+                        s.unlock_conditions().as_ref(),
+                        s.features().as_ref(),
+                    ) {
+                        // all fine, continue with next output
+                        continue;
+                    }
+                }
+                // not fine, return
+                return true;
+            }
+        }
+    }
+    // check if essence + bip32 indices fit into the buffer of the device
+    let essence_bytes = prepared_transaction.essence.pack_to_vec();
+    let total_size =
+        LedgerBIP32Index::default().packed_len() * prepared_transaction.inputs_data.len() + essence_bytes.len();
+
+    // return true if too large
+    total_size > buffer_size
+}
+
 #[async_trait]
 impl SecretManageExt for LedgerSecretManager {
     async fn sign_transaction_essence(&self, prepared_transaction: &PreparedTransactionData) -> crate::Result<Unlocks> {
         // lock the mutex to prevent multiple simultaneous requests to a ledger
         let _lock = self.mutex.lock().await;
 
-        // Get coin type and account index from first input
-        let (_coin_type, account_index) = match &prepared_transaction.inputs_data.first() {
-            Some(input) => {
-                match &input.chain {
-                    Some(chain) => {
-                        let raw: Vec<u32> = chain
-                            .segments()
-                            .iter()
-                            // XXX: "ser32(i)". RTFSC: [crypto::keys::slip10::Segment::from_u32()]
-                            .map(|seg| u32::from_be_bytes(seg.bs()))
-                            .collect();
-                        (raw[0], raw[1])
+        let mut input_bip32_indices: Vec<LedgerBIP32Index> = Vec::new();
+        let mut coin_type: Option<u32> = None;
+        let mut account_index: Option<u32> = None;
+
+        let input_len = prepared_transaction.inputs_data.len();
+
+        for input in &prepared_transaction.inputs_data {
+            let bip32_indices: Vec<u32> = match &input.chain {
+                Some(chain) => {
+                    chain
+                        .segments()
+                        .iter()
+                        // XXX: "ser32(i)". RTFSC: [crypto::keys::slip10::Segment::from_u32()]
+                        .map(|seg| u32::from_be_bytes(seg.bs()))
+                        .collect()
+                }
+                None => return Err(crate::Error::NoInputs),
+            };
+            // coin_type and account_index should be the same in each output
+            if (coin_type.is_some() && coin_type != Some(bip32_indices[0]))
+                || (account_index.is_some() && account_index != Some(bip32_indices[1]))
+            {
+                return Err(crate::Error::InvalidBIP32ChainData);
+            }
+
+            coin_type = Some(bip32_indices[1]);
+            account_index = Some(bip32_indices[2]);
```

Comment:
Is this `1` and `2` ? Because in the `if` it was `0` and `1`

Reply:
ups, nice catch ... I guess we only tried with one single input :see_no_evil: